client/src/views/devices/ImportRecoveryDevicePage.vue

```diff
@@ -0,0 +1,261 @@
+import { MsChoosePasswordInput, MsInformativeText, MsInput, MsReportText, MsReportTheme } from '@/components/core';
+import { AvailableDevice, RecoveryImportErrorTag, createRecoveredDevice, importRecoveryDevice } from '@/parsec';
+import { Notification, NotificationKey, NotificationLevel, NotificationManager } from '@/services/notificationManager';
+import { IonButton, IonCard, IonCardContent, IonCardTitle, IonIcon, IonText } from '@ionic/vue';
+import { checkmarkCircle, document, key } from 'ionicons/icons';
+import { Ref, defineEmits, defineProps, inject, onMounted, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+const { t } = useI18n();
+
+enum ImportDevicePageState {
+  Start = 'start',
+  Password = 'password',
+  Done = 'done',
+}
+
+const state = ref(ImportDevicePageState.Start);
+const choosePasswordInput: Ref<typeof MsChoosePasswordInput | null> = ref(null);
+const hiddenInput = ref();
+const secretKey = ref('');
+// cspell:disable-next-line
+const secretKeyPlaceholder = '7BWT-WG5M-5BIY-TE7H-A4HX-67ZD-WTNB-22ZV-VJPT-D3Z3-5UWP-HGFL-7J3A';
+const changeButtonIsEnabled = asyncComputed(async (): Promise<boolean> => {
+  return choosePasswordInput.value && (await choosePasswordInput.value.areFieldsCorrect());
+});
+const recoveryFile: Ref<File | null> = ref(null);
+const isSecretKeyValid = ref(false);
+const notificationManager: NotificationManager = inject(NotificationKey)!;
+
+const emits = defineEmits<{
+  (e: 'finished'): void;
+}>();
+
+const props = defineProps<{
+  device: AvailableDevice;
+}>();
+
+onMounted(() => {
+  hiddenInput.value.addEventListener('change', onInputChange);
+});
+
+async function onInputChange(_event: Event): Promise<void> {
+  if (hiddenInput.value.files.length === 1) {
+    recoveryFile.value = hiddenInput.value.files[0];
+  }
+}
+
+async function importButtonClick(): Promise<void> {
+  hiddenInput.value.click();
+}
+
+async function checkSecretKeyValidity(): Promise<void> {
+  isSecretKeyValid.value = (await secretKeyValidator(secretKey.value)) === Validity.Valid;
+}
+
+async function goToPasswordChange(): Promise<void> {
+  if (recoveryFile.value) {
+    const result = await importRecoveryDevice(recoveryFile.value, secretKey.value);
+    if (result.ok) {
+      state.value = ImportDevicePageState.Password;
+    } else {
+      if (result.error.tag === RecoveryImportErrorTag.KeyError) {
+        notificationManager.showToast(
+          new Notification({
+            message: t('ImportRecoveryDevicePage.errors.keyError'),
+            level: NotificationLevel.Error,
+          }),
+        );
+      }
+      if (result.error.tag === RecoveryImportErrorTag.RecoveryFileError) {
+        notificationManager.showToast(
+          new Notification({
+            message: t('ImportRecoveryDevicePage.errors.fileError'),
+            level: NotificationLevel.Error,
+          }),
+        );
+      }
+    }
+  }
+}
+
+async function createNewDevice(): Promise<void> {
+  if (changeButtonIsEnabled.value) {
```

Comment:
Don't rely on the state of a button. Check manually that you have the information you need to progress.
Simple states are already a mess by themselves, if you have a state that depends on another state that depends on another state, it is a huge source of bugs.